platform/darwin/src/NSCoder+MGLAdditions.mm

```diff
@@ -0,0 +1,16 @@
+#import "NSCoder+MGLAdditions.h"
+
+#import "NSArray+MGLAdditions.h"
+
+@implementation NSCoder (MGLAdditions)
+
+- (void)mgl_encodeLocationCoordinates2D:(std::vector<CLLocationCoordinate2D>)coordinates forKey:(NSString *)key {
+    [self encodeObject:[NSArray mgl_coordinatesFromCoordinates:coordinates] forKey:key];
```

Comment:
I use NSKeyedArchiver for testing NSCoding which can't encode structs, not even structs wrapped in NSValue encoded with `@encode(CLLocationCoordinate2D)`. (```[NSKeyedArchiver encodeValueOfObjCType:at:]: this archiver cannot encode structs```) Any other idea how to solve this? UIStateRestoring will probably still work if we switch to NSValue but a solution that works with archiving as well would be preferred.